src/components/ChecSlideoutPanel/Overlay.vue

```diff
@@ -0,0 +1,60 @@
+<template>
+  <div
+    :class="`slideout-panel__overlay slideout-panel__overlay--${overlay}`"
+    @click="onOverlayDismiss"
+  />
+</template>
+
+<script>
+export default {
+  name: 'Overlay',
+  props: {
+    /**
+     * The overlay backdrop when the panel is opened. One of "light" and "dark".
+     */
+    overlay: {
+      type: String,
+      default: 'light',
+      validator(value) {
+        return ['light', 'dark'].includes(value);
+      },
+    },
+    /**
+     * Whether to close panel on overlay click. Default is true.
+     */
+    closeOnOverlayClick: Boolean,
```

Comment:
The parent component has this prop already. I think this component can emit the event regardless, and then the parent component can check this prop and decide whether to do anything with the event.